linera-core/src/chain_worker/state/mod.rs

```diff
@@ -210,13 +211,18 @@ where
             .check_invariants()
             .map_err(|msg| WorkerError::InvalidBlockProposal(msg.to_string()))?;
         proposal.check_signature()?;
-        ChainWorkerStateWithAttemptedChanges::new(&mut *self)
+        let Some(published_blobs) = ChainWorkerStateWithAttemptedChanges::new(&mut *self)
             .await
             .validate_block(&proposal)
-            .await?;
+            .await?
+        else {
+            // Skipping: We already voted for this block.
+            let info = ChainInfoResponse::new(&self.chain, self.config.key_pair());
+            return Ok((info, NetworkActions::default()));
+        };
```

Comment:
I don't think it's a good design where `validate_block` returns `Option<published_blobs>` and when that's `None` it means we already voted for this block. That's at least two separate responsibilities mixed in one method.